sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/ObjectAggregationIterator.scala

```diff
@@ -252,6 +249,7 @@ class SortBasedAggregator(
       var hasNextAggBuffer: Boolean = initialAggBufferIterator.next()
       private var result: AggregationBufferEntry = _
       private var groupingKey: UnsafeRow = _
+      private var aggregateMode: Int = _
```

Comment:
Sort based aggregate has no such code. The difference is because the object hash aggreagtion has two iterator. One is a input buffer which is generated before fallback to sort based and the other is input rows. Then after going to sort based, we should do update for input rows and do merge for input buffer in partial mode.

This variable is used to avoid unnecessary grouping key comparation.

Reply:
Since this is a new code branch, can we add detailed comments to explain all the states and how state transition happens?